### PR TITLE
Enhance documentation

### PR DESCRIPTION
**Pull request by SunSince90**
This pull request completely re-imagines and enhances documentation.

Documentation is now split into chapters/sections inside `docs` folder.
A new Quickstart is created for people who want to quickly see the operator in action, or just feel lazy reading documentation :)

Finally, Readme is updated to be much slimmer and it does not scare users anymore. It also features some new shiny badges on top :).

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+To run this, make sure you have:
+
+* Access to a Kubernetes cluster running at least version `1.11.3`
+  * [Minikube](https://kubernetes.io/docs/setup/learning-environment/minikube/)
+  is also fine.
+* [Kubectl 1.11.3+](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+* A Project in [Google Cloud](https://console.cloud.google.com/) with
+[Service Directory](https://cloud.google.com/service-directory) enabled
+* A [Google Cloud Service Account](https://cloud.google.com/iam/docs/service-accounts)
```
Comment (ljakab) [inline docs/quickstart.md:16]: We need to specify the minimum roles the service accounts needs. Is it Service Directory Admin (`roles/servicedirectory.admin`) ?

**Review by ljakab — COMMENTED** [review]
I'm going to post small separate comments while going through the docs, to spread out the review.

File: docs/quickstart.md (hunk 1 above)
Comment (SunSince90) [reply docs/quickstart.md:16]: Good catch! Most probably not, but I will look into it, thank you!

File: README.md [hunk 2]
```diff
@@ -47,362 +43,20 @@ account file must be provided as a `Secret`.
...
 
-To remove the operator from your Kubernetes cluster, navigate to the root
-directory of the project and execute:
+If you want to learn more about CNWAN Operator, you can follow the
+documentation inside the [docs](./docs) folder.
 
-```bash
-make remove
-```
+You can start by reading [Concepts](./docs/concepts.md) or try to deploy it
```
Comment (arnatal) [inline README.md:56]: Maybe instead of this sentence, can we have bullet points in the "Documentation" section that describe each of the files available in the ./docs folder?

File: docs/advanced_installation.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  brew install kubebuilder
+  ```
+
+  * *Other systems*: follow
+  [this section](https://book.kubebuilder.io/quick-start.html#installation)
+  on the documentation.
+
+### Your files
+
+You will need to place files depending on what kind of files you want to modify.
```
Comment (arnatal) [inline docs/advanced_installation.md:77]: Not sure I understand this sentence. In general the "Your files" section it's a bit confusing to me (it may be just me)

Comment (SunSince90) [reply docs/advanced_installation.md:77]: That's for adding resources for advanced users. For example you may add new Roles, Deployments, Services etc. Depending on the kind of the resource you need to place the file on the appropriate folder, so roles go to rbac folder and deployments/services go to manager and so on (but mostly users will want to place them in these two folders).

People that go this way already know about kubernetes & operators for the most part, so that's why I did not elaborate it too much.

File: docs/quickstart.md (hunk 1 above)
Comment (SunSince90) [reply docs/quickstart.md:16]: I specified the roles needed. Thank you!

File: README.md (hunk 2 above)
Comment (SunSince90) [reply README.md]: Just did, although I forgot to write description. I think though that the names are understandable. Let me know if you think it's better to change them though

File: docs/advanced_installation.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+* access to a Kubernetes cluster running at least version `1.11.3` and
+[kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/)
+properly set up
+* a Project in [Google Cloud](https://console.cloud.google.com/) with
+[Service Directory](https://cloud.google.com/service-directory) enabled
+* a [Google Cloud Service Account](https://cloud.google.com/iam/docs/service-accounts)
+with role `roles/servicedirectory.editor` or `roles/servicedirectory.admin`.
+
+### Programs
```
Comment (arnatal) [inline docs/advanced_installation.md:24]: Shall we call this "Software"?

File: docs/advanced_installation.md (hunk 3 above)
Comment (arnatal) [reply docs/advanced_installation.md:78]: Shall we refer to these as "custom files" then? Is that name accurate?

Comment (SunSince90) [reply docs/advanced_installation.md:78]: I think that custom files would mostly remind of crds. I'll see how I can reword this

Comment (SunSince90) [reply docs/advanced_installation.md]: I rephrased the whole section and also specified to not follow the guide if you're not touch Kubernetes resources.
I am actually thinking about removing those files and the dependency from Kustomize entirely. I will have to think about this a bit.

File: docs/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+with role `roles/servicedirectory.editor` or `roles/servicedirectory.admin`.
```
Comment (ljakab) [inline docs/quickstart.md:17]: Same comment as in the basic install doc

**Review by ljakab — CHANGES_REQUESTED** [review]
Excellent documentation @SunSince90  thank you very much for the write-up. I have a few comments inline.

File: README.md [hunk 6]
```diff
@@ -12,28 +19,17 @@ You can contact the CNWAN team at [cnwan@cisco.com](mailto:cnwan@cisco.com).
...
-    * [Google Cloud Service Account](#google-cloud-service-account)
-* [Deploying to Your Cluster](#deploying-to-your-cluster)
-  * [Kubernetes Requirements](#kubernetes-requirements)
-  * [Before Deploying](#before-deploying)
-  * [Deploy](#deploy)
-* [Removing from Your Cluster](#removing-from-your-cluster)
+* [Try It Out](#try-it-out)
+* [Documentation](#documentation)
 * [Contributing](#contributing)
 * [License](#license)
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply README.md:25]: Good point!

File: README.md [hunk 7]
```diff
@@ -47,362 +43,17 @@ account file must be provided as a `Secret`.
...
-directory of the project and execute:
+## Documentation
 
-```bash
-make remove
-```
+* [Concepts](./concepts.md)
+* [Configuration](./configuration.md)
+* [Basic Installation](./basic_installation.md)
+* [Advanced Installation](./advanced_installation.md)
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply README.md:56]: Whoops, thank you!

File: docs/basic_installation.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
+# Basic Installation
+
+This is the easiest way to install the operator and suitable for most users.
+
+This will install/deploy the operator with the most recent version and without
+changing any resource created by the CNWAN Owners.
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply docs/basic_installation.md:6]: I can probably rephrased this `yaml resources` to make it better understandable.

File: docs/basic_installation.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+You need to have the following:
+
+* access to a Kubernetes cluster running at least version `1.11.3` and
+[kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/)
+properly set up
+* a Project in [Google Cloud](https://console.cloud.google.com/) with
+[Service Directory](https://cloud.google.com/service-directory) enabled
+* a [Google Cloud Service Account](https://cloud.google.com/iam/docs/service-accounts)
+with role `roles/servicedirectory.editor` or `roles/servicedirectory.admin`.
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply docs/basic_installation.md:20]: For the operator `editor` is enough. Maybe I can rephrase this as just "at least `editor` permissions level" (or something along these lines)

File: docs/configuration.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+the CNWAN Operator.
+
+## Table of Contents
+
+* [Settings](#settings)
+* [Google Cloud Settings](#google-cloud-settings)
+* [Set the Namespace List Policy](#set-the-namespace-list-policy)
+* [Allow Annotations](#allow-annotations)
+
+## Settings
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply docs/configuration.md:13]: Thank you, these are something I brought from the previous version (I rewrote them later on) and didn't change accordingly

File: docs/basic_installation.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (ljakab) [reply docs/basic_installation.md:20]: Sounds good, I think it's important to always indicate the minimum required for the most security conscious users

File: docs/basic_installation.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ljakab) [reply docs/basic_installation.md:6]: Ah, ok. In that case, to make it extra clear, I would not mention "CNWAN Owners", I would just say "install/deploy the latest released version of the operator with the unmodified YAML resources from this repository." I wanted to add ** released** version, because the master can be thought of as version too, which is a moving target.

File: docs/advanced_installation.md (hunk 4 above)
Comment (SunSince90) [reply docs/advanced_installation.md]: Changed in both docs :)

File: docs/basic_installation.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (SunSince90) [reply docs/basic_installation.md:6]: Yes, I agree. It makes more sense like that

**Comment by arnatal** [comment]
This looks fantastic, but I have the feeling that we are missing some extra examples aligned with the rest of the CNWAN project. For instance, having an example with "cnwan.io/traffic-profile" in the configuration file. However, I'm leaning towards merging this PR and opening an issue to keep track of that. Thanks for the great work on the new documentation Elis!
